ruby/BUILDING.md

```diff
@@ -1,172 +1,85 @@
-Ruby installation's `site_ruby` directory. For example, if you installed Ruby
-via RPM, you can use the steps below:
-
-```shell
-cd <Ice source directory>/ruby/ruby
-sudo tar cf - * | (cd /usr/lib/ruby/site_ruby/1.8/i386-linux; tar xvf -)
-```
-
-On x86_64 systems, change the last command to:
-
-```shell
-sudo tar cf - * | (cd /usr/lib64/ruby/site_ruby/1.8/x86_64-linux; tar xvf -)
-```
-
-There is no need to modify your environment if you use this approach.
-
-## Configuring your Environment for Ruby
-
-The Ruby interpreter must be able to locate the Ice extension. If you used the
-automated installation described above, you need to define the `RUBYLIB`
-environment variable as follows:
+### Building Ice for Ruby
 
-```shell
-export RUBYLIB=/opt/Ice/ruby:$RUBYLIB
-```
-
-This example assumes that your Ice for Ruby installation is located in the
-`/opt/Ice` directory.
-
-You must also modify `LD_LIBRARY_PATH` or `DYLD_LIBRARY_PATH` to include the
-directory `/opt/Ice/lib`:
-
-```shell
-export LD_LIBRARY_PATH=/opt/Ice/lib:$LD_LIBRARY_PATH       (Linux)
-export DYLD_LIBRARY_PATH=/opt/Ice/lib:$DYLD_LIBRARY_PATH   (macOS)
-```
-
-To verify that Ruby can load the Ice extension successfully, open a command
-window and start the interpreter using `irb`.
+Before building Ice for Ruby, you must first build the Ice for C++ source distribution.
+Refer to the [build instructions](../cpp/BUILDING.md) in the `cpp` subdirectory for details.
 
-At the prompt, enter:
+Once Ice for C++ is built, open a terminal and navigate to the `ruby` subdirectory.
+To build Ice for Ruby, run:
 
 ```shell
-require "Ice"
+make
 ```
 
-If the interpreter responds with the value true, the Ice extension was loaded
-successfully. Enter `exit` to quit the interpreter.
-
-## Running the Ruby Tests
-
-The `test` subdirectory contains Ruby implementations of the core Ice test
-suite. Python is required to run the test suite.
+### Running the Tests
 
-The test suites require that the Ice for C++ tests be built in the `cpp`
-subdirectory of this source distribution.
-
-Open a command window and change to the top-level directory. At the command
-prompt, execute:
-
-```shell
-python allTests.py
-```
-
-You can also run tests individually by changing to the test directory and
-running this command:
+To run the tests, open a terminal and change to the `ruby` subdirectory. Then run:
 
 ```shell
-python run.py
+python allTests.py --all
 ```
 
-If everything worked out, you should see lots of `ok` messages. In case of a
-failure, the tests abort with `failed`.
+### Generating the API Reference
 
-## SELinux Notes for Ruby
-
-If SELinux is enabled on your RHEL system, you may encounter this error message
-when Ruby attempts to load the Ice extension:
+To build the API reference documentation, run the following command from the `ruby` subdirectory:
 
 ```shell
-cannot restore segment prot after reloc: Permission denied
+yard doc 'ruby/**/*.rb'
```

Comment:
This is fine for now. Our documentation is currently very incomplete. We'll need to update this in the future.